Description: Update src/tag.cc for TagLib >= 1.12.
 Adapted from an upstream commit.
Author: Hugh McMaster <hmc@debian.org>
Origin: https://github.com/Moonbase59/loudgain/commit/0e03353e6d90198204c2edd9ccf9ecaad6a8e7c8
Forwarded: not-needed
Last-Update: 2024-08-12

--- a/src/tag.cc
+++ b/src/tag.cc
@@ -204,7 +204,9 @@
   if (strip)
     f.strip(TagLib::MPEG::File::APE);
 
-  return f.save(TagLib::MPEG::File::ID3v2, strip, id3v2version);
+  return f.save(TagLib::MPEG::File::ID3v2,
+    strip ? TagLib::MPEG::File::StripOthers : TagLib::MPEG::File::StripNone,
+    id3v2version == 3 ? TagLib::ID3v2::v3 : TagLib::ID3v2::v4);
 }
 
 bool tag_clear_mp3(scan_result *scan, bool strip, int id3v2version) {
@@ -217,7 +219,9 @@
   if (strip)
     f.strip(TagLib::MPEG::File::APE);
 
-  return f.save(TagLib::MPEG::File::ID3v2, strip, id3v2version);
+  return f.save(TagLib::MPEG::File::ID3v2,
+    strip ? TagLib::MPEG::File::StripOthers : TagLib::MPEG::File::StripNone,
+    id3v2version == 3 ? TagLib::ID3v2::v3 : TagLib::ID3v2::v4);
 }
 
 
@@ -490,9 +494,9 @@
 
 void tag_remove_mp4(TagLib::MP4::Tag *tag) {
   TagLib::String desc;
-  TagLib::MP4::ItemListMap &items = tag->itemListMap();
+  TagLib::MP4::ItemMap items = tag->itemMap();
 
-  for(TagLib::MP4::ItemListMap::Iterator item = items.begin();
+  for(TagLib::MP4::ItemMap::Iterator item = items.begin();
       item != items.end(); ++item)
   {
     desc = item->first.upper();
@@ -715,7 +719,9 @@
   }
 
   // no stripping
-  return f.save(TagLib::RIFF::WAV::File::AllTags, false, id3v2version);
+  return f.save(TagLib::RIFF::WAV::File::AllTags,
+    TagLib::RIFF::WAV::File::StripNone,
+    id3v2version == 3 ? TagLib::ID3v2::v3 : TagLib::ID3v2::v4);
 }
 
 bool tag_clear_wav(scan_result *scan, bool strip, int id3v2version) {
@@ -725,7 +731,9 @@
   tag_remove_wav(tag);
 
   // no stripping
-  return f.save(TagLib::RIFF::WAV::File::AllTags, false, id3v2version);
+  return f.save(TagLib::RIFF::WAV::File::AllTags,
+    TagLib::RIFF::WAV::File::StripNone,
+    id3v2version == 3 ? TagLib::ID3v2::v3 : TagLib::ID3v2::v4);
 }
 
 
